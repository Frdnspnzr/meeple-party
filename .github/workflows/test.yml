name: Test
on:
  pull_request:
    branches:
      - "*"
  workflow_dispatch:
    inputs:
      update-snapshots:
        description: "Update snapshots?"
        type: boolean
permissions:
  contents: read
  pages: write
  id-token: write
jobs:
  build:
    timeout-minutes: 10
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-node@v4
        with:
          node-version: lts/*
          cache: "npm"
      - name: Install dependencies
        run: npm install
      - name: Generate database bindings
        run: npm run "prisma generate"
      - name: Build app
        run: npm run build
      - name: Save build
        uses: actions/upload-artifact@v4
        with:
          name: build
          path: .next

  e2e:
    timeout-minutes: 60
    runs-on: ubuntu-latest
    needs: build
    services:
      postgres:
        image: postgres
        env:
          POSTGRES_PASSWORD: postgres
        options: >-
          --health-cmd pg_isready
          --health-interval 10s
          --health-timeout 5s
          --health-retries 5
        ports:
          - 5432:5432
    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-node@v4
        with:
          node-version: lts/*
          cache: "npm"
      - name: Install dependencies
        run: npm install
      - name: Install Playwright browsers
        run: npx playwright install --with-deps
      - name: Load build
        uses: actions/download-artifact@v4
        with:
          name: build
          path: .next
      - name: Set up snapshot cache
        id: snapshots_cache
        uses: actions/cache@v4
        with:
          key: ${{ runner.os }}-snapshots-${{github.ref}}
          restore-keys: ${{ runner.os }}-snapshots-refs/heads/main
          path: ./tests/e2e/snapshots/**
      - name: Update snapshots
        if: ${{steps.snapshots_cache.outputs.cache-hit != 'true' || inputs.update-snapshots == 'true'}}
        run: npx playwright test --update-snapshots
      - name: Generate database bindings
        run: npm run "prisma generate"
      - name: Generate database tables
        run: npm run "prisma deploy"
        env:
          DATABASE_URL: postgresql://postgres:postgres@localhost:5432/test
      - name: Run Playwright tests
        run: npx playwright test --max-failures=10
        env:
          DATABASE_URL: postgresql://postgres:postgres@localhost:5432/test
          NEXT_PUBLIC_SECRET: Testrunner
          NEXTAUTH_URL: http://127.0.0.1:3000
      - uses: actions/upload-artifact@v4
        if: always()
        with:
          name: playwright-report
          path: playwright-report/

  unittest:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - name: Setup Node.js
        uses: actions/setup-node@v3
        with:
          node-version: lts/*
          cache: "npm"
      - name: Install dependencies
        run: npm install
      - name: Generate database bindings
        run: npm run "prisma generate"
      - name: Run the tests
        run: npm test
      - name: Upload coverage reports to Codecov
        uses: codecov/codecov-action@v3
        env:
          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}

  publish-e2e:
    runs-on: ubuntu-latest
    needs: [e2e, build]
    if: ${{ !cancelled() && (needs.build.result=='success') }}
    steps:
      - name: Configure GitHub Pages
        uses: actions/configure-pages@v3
      - name: Load report
        uses: actions/download-artifact@v4
        with:
          name: playwright-report
          path: playwright-report/
      - name: Upload Pages Artifact
        uses: actions/upload-pages-artifact@v2
        with:
          path: playwright-report/
      - name: Deploy to GitHub Pages
        uses: actions/deploy-pages@v2
