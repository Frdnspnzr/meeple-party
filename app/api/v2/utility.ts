import { Relationship, RelationshipType } from "@/datatypes/relationship";
import {
  UserPreferences,
  UserProfile,
  defaultUserPreferences,
} from "@/datatypes/userProfile";
import {
  Prisma,
  Relationship as PrismaRelationship,
  User,
} from "@prisma/client";

export type FullPrismaRelationship = PrismaRelationship & {
  sender: User;
  recipient: User;
};

export function convertToUserProfile(
  user: User,
  friend: boolean = false
): UserProfile {
  return {
    id: user.id,
    name: user.name,
    role: user.role,
    image: user.image,
    realName: allowRealName(user, friend) ? user.realName : null,
    about: user.about,
    place: allowPlace(user, friend) ? user.place : null,
    bggName: allowBggName(user, friend) ? user.bggName : null,
  };
}

export function normalizeRelationship(
  prismaRelationship: FullPrismaRelationship,
  userId: string
): Relationship {
  return {
    profile: getProfile(prismaRelationship, userId),
    type: getRelationshipType(prismaRelationship, userId),
    lastUpdate: prismaRelationship.updatedAt,
  };
}

function allowRealName(user: User, friend: boolean): boolean {
  if (friend) {
    return true;
  }
  const preferences = getPreferences(user);
  return preferences.showRealNameInProfile;
}

function allowPlace(user: User, friend: boolean): boolean {
  if (friend) {
    return true;
  }
  const preferences = getPreferences(user);
  return preferences.showPlaceInProfile;
}

function allowBggName(user: User, friend: boolean): boolean {
  return friend;
}

function getPreferences(user: User): UserPreferences {
  return {
    ...defaultUserPreferences,
    ...(user.preferences as Prisma.JsonObject),
  };
}

function getRelationshipType(
  relationship: FullPrismaRelationship,
  userId: string
): RelationshipType {
  if (relationship.type === "FRIENDSHIP") {
    return RelationshipType.FRIENDSHIP;
  } else if (relationship.type === "FRIEND_REQUEST") {
    if (relationship.sender.id === userId) {
      return RelationshipType.FRIEND_REQUEST_SENT;
    } else {
      return RelationshipType.FRIEND_REQUEST_RECEIVED;
    }
  }
  throw new Error("Error determining relationship type");
}

function getProfile(
  relationship: FullPrismaRelationship,
  userId: string
): UserProfile {
  const profile =
    relationship.recipientId === userId
      ? relationship.sender
      : relationship.recipient;
  return convertToUserProfile(profile, relationship.type === "FRIENDSHIP");
}

/**
 * Reads the authentication token for cronjobs from the environment variable `CRON_AUTH_TOKEN`. Throws when that variable is not set.
 * @returns Authentication token.
 * @throws Throws when no token is configured.
 */
export function getCronAuthToken() {
  if (!process.env.CRON_AUTH_TOKEN) {
    throw new Error("No cron auth token configured for current environment");
  }
  return process.env.CRON_AUTH_TOKEN;
}
