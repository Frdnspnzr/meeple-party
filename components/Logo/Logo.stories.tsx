import type { Meta, StoryObj } from "@storybook/react";
import Logo from "./Logo";

const meta = {
  title: "Parts/Logo",
  component: Logo,
  tags: ["autodocs"],
} satisfies Meta<typeof Logo>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Large: Story = {
  args: {
    size: "lg",
  },
};

export const Medium: Story = {
  args: {
    size: "md",
  },
};

export const Small: Story = {
  args: {
    size: "sm",
  },
};

export const ExtraSmall: Story = {
  args: {
    size: "xs",
  },
};
