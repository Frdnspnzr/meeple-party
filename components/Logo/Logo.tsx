import Image from "next/image";
import svg from "./logo.svg";

type Size = "xs" | "sm" | "md" | "lg";

export interface LogoProps {
  size: Size;
  className?: string;
  unstyled?: boolean;
}

export default function Logo({
  size,
  className = "",
  unstyled = false,
}: Readonly<LogoProps>) {
  const widthToLoad = getWidthToLoad(size);
  const heightToLoad = getHeightToLoad(widthToLoad);
  const height = getHeight(size);
  return (
    <Image
      src={svg}
      width={widthToLoad}
      height={heightToLoad}
      alt="Meeple Party"
      style={unstyled ? {} : { height, width: "100%" }}
      className={className}
    />
  );
}

function getWidthToLoad(size: Size): number {
  switch (size) {
    case "xs":
      return 16;
    case "sm":
      return 24;
    case "md":
      return 96;
    case "lg":
      return 400;
  }
}

function getHeightToLoad(width: number): number {
  return (width * 3) / 4;
}

function getHeight(size: Size): string {
  switch (size) {
    case "xs":
      return "2ex";
    case "sm":
      return "3rem";
    case "md":
      return "6rem";
    case "lg":
      return "18rem";
  }
}
