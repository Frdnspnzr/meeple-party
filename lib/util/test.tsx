import {
  generateArray,
  generateBoolean,
  generateDate,
  generateGame,
} from "@/utility/test";
import { Game as ServerGame } from "@prisma/client";
import { Collection } from "../datatypes/client/collection";
import { Game } from "../datatypes/client/game";
import { UserId } from "../datatypes/client/userProfile";

export function ObjectData({ object }: { object: object }) {
  return (
    <>
      {Object.entries(object).map(([key, value]) => (
        <div data-testid={key} key={key}>
          {JSON.stringify(value)}
        </div>
      ))}
    </>
  );
}

export function generateCollection(userId: UserId): {
  collection: Collection;
  games: Game[];
} {
  const games = generateArray(generateGame);
  const randomGames = () => pickRandom(games).map((g) => g.id);
  const collection = {
    user: userId,
    own: randomGames(),
    wantToPlay: randomGames(),
    wishlist: randomGames(),
  };

  return {
    games,
    collection,
  };
}

export function convertClientGame(game: Game): ServerGame {
  return {
    ...game,
    thumbnail: game.thumbnail || null,
    image: game.image || null,
    BGGRank: game.BGGRank || null,
    BGGRating: game.BGGRating || null,
    updatedAt: generateDate(),
  };
}

export function pickRandom<T>(arr: T[]): T[] {
  return arr.filter(generateBoolean);
}
