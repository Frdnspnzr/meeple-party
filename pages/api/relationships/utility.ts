
import { Relationship, RelationshipType } from "@/datatypes/relationship";
import { PrivateUserProfile, PublicUserProfile } from "@/datatypes/userProfile";
import { prisma } from "@/db";
import { Session } from "@auth0/nextjs-auth0";
import { UserProfile as Auth0Profile } from "@auth0/nextjs-auth0/client";
import {
  Relationship as PrismaRelationship,
  UserProfile,
} from "@prisma/client";

export type FullPrismaRelationship = PrismaRelationship & {
  sender: UserProfile;
  recipient: UserProfile;
};

export function normalizeRelationship(
  prismaRelationship: FullPrismaRelationship,
  userId: number
): Relationship {
  return {
    profile: getProfile(prismaRelationship, userId),
    type: getRelationshipType(prismaRelationship, userId),
    lastUpdate: prismaRelationship.updatedAt,
  };
}

export function getRelationshipType(
  relationship: FullPrismaRelationship,
  userId: number
): RelationshipType {
  if (relationship.type === "FRIENDSHIP") {
    return RelationshipType.FRIENDSHIP;
  } else if (relationship.type === "FRIEND_REQUEST") {
    if (relationship.sender.id === userId) {
      return RelationshipType.FRIEND_REQUEST_SENT;
    } else {
      return RelationshipType.FRIEND_REQUEST_RECEIVED;
    }
  }
  throw new Error("Error determining relationship type");
}

export function getProfile(
  relationship: FullPrismaRelationship,
  userId: number
): PrivateUserProfile | PublicUserProfile {
  const profile =
    relationship.recipientId === userId
      ? relationship.sender
      : relationship.recipient;
  if (
    relationship.type === "FRIENDSHIP" ||
    isFriendRequestReveiced(relationship, userId)
  ) {
    return convertToPrivateProfile(profile);
  } else {
    return convertToPublicProfile(profile);
  }
}

export function convertToPrivateProfile(profile: UserProfile): PrivateUserProfile {
  return {
    ...convertToPublicProfile(profile),
    realName: profile.realName,
  };
}

export function convertToPublicProfile(profile: UserProfile): PublicUserProfile {
  return {
    id: profile.id,
    name: profile.name,
    picture: profile.picture,
  };
}

export function isFriendRequestReveiced(
  relationship: FullPrismaRelationship,
  userId: number
): boolean {
  return (
    relationship.type === "FRIEND_REQUEST" &&
    relationship.recipientId === userId
  );
}

export async function getUserProfile(session: Session): Promise<UserProfile> {
  const user = session.user as Auth0Profile;
  if (!user.email) {
    throw new Error("User from Auth0 does not have an E-Mail adress");
  }
  const userProfile = await prisma.userProfile.findUnique({
    where: { email: user.email as string },
  });

  if (!userProfile) {
    throw new Error("No user profile corresponding to Auth0 user found.");
  }

  return userProfile;
}