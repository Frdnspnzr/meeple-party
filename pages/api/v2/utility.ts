import { authOptions } from "@/pages/api/auth/[...nextauth]";
import { User } from "@prisma/client";
import { NextApiRequest, NextApiResponse } from "next";
import { getServerSession } from "next-auth/next";
import { ApiError } from "./types";

type ApiHandler = (req: NextApiRequest, res: NextApiResponse) => Promise<void>;
type ApiHandlerWithUser = (
  req: NextApiRequest,
  res: NextApiResponse,
  user: User
) => Promise<void>;

export function withUser(handler: ApiHandlerWithUser): ApiHandler {
  return async (req, res) => {
    const session = await getServerSession(req, res, authOptions);
    if (!session || !session.user) {
      throw Error("Unauthorized");
    }
    return await handler(req, res, session.user);
  };
}

export function withErrorHandling(handler: ApiHandler): ApiHandler {
  return async (req, res) => {
    try {
      return await handler(req, res);
    } catch (reason) {
      res.status(401).send({ reason } as ApiError);
    }
  };
}
