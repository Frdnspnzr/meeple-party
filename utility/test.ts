import { FullPrismaRelationship } from "@/app/api/v2/utility";
import { defaultUserPreferences } from "@/datatypes/userProfile";
import { RelationshipType, Role, User } from "@prisma/client";

test("Foo", () => {});

export function generateString(length = 32) {
  let result = "";
  const characters =
    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
  while (result.length < length) {
    result += characters.charAt(Math.floor(Math.random() * characters.length));
  }
  return result;
}

export function generateArray<T>(generator: () => T, length = 10): T[] {
  const ret: T[] = [];
  for (let i = 0; i < length; i++) {
    ret.push(generator());
  }
  return ret;
}

export function generatePrismaUser(): User {
  return {
    id: generateString(25),
    name: generateString(25),
    email: generateString(25),
    profileComplete: true,
    role: Role.USER,
    emailVerified: new Date(),
    about: generateString(1000),
    bggName: generateString(15),
    image: null,
    place: generateString(15),
    realName: generateString(25),
    preferences: defaultUserPreferences,
  };
}

export function generateFullPrismaRelationship(
  type: RelationshipType = RelationshipType.FRIENDSHIP,
  userA?: User,
  userB?: User
): FullPrismaRelationship {
  if (!userA) {
    userA = generatePrismaUser();
  }
  if (!userB) {
    userB = generatePrismaUser();
  }

  return {
    senderId: userA.id,
    recipientId: userB.id,
    type: type,
    sender: userA,
    recipient: userB,
    updatedAt: new Date(),
    cratedAt: new Date(),
  };
}
