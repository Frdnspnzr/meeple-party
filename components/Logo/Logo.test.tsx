import { render } from "@/utility/test";
import Logo from "./Logo";

describe("Profile badge", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });
  test.each(["xs", "sm", "md", "lg"])("size %s matches snapshot", (size) => {
    const { container } = render(<Logo size={size as any} />);
    expect(container).toMatchSnapshot();
  });
  test.each(["xs", "sm", "md", "lg"])(
    "size %s matches unstyled snapshot",
    (size) => {
      const { container } = render(<Logo size={size as any} unstyled />);
      expect(container).toMatchSnapshot();
    }
  );
  test.each(["xs", "sm", "md", "lg"])(
    "size %s matches snapshot with additional class names",
    (size) => {
      const className = "test class name";
      const { container } = render(
        <Logo size={size as any} className={className} />
      );
      expect(container).toMatchSnapshot();
    }
  );
});
