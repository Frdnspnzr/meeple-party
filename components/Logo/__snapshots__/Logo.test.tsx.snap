// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile badge size lg matches snapshot 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <img
    alt="Meeple Party"
    class=""
    data-nimg="1"
    decoding="async"
    height="300"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=828&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg.jpg&w=828&q=75 2x"
    style="color: transparent; height: 18rem; width: 100%;"
    width="400"
  />
</div>
`;

exports[`Profile badge size lg matches snapshot with additional class names 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <img
    alt="Meeple Party"
    class="test class name"
    data-nimg="1"
    decoding="async"
    height="300"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=828&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg.jpg&w=828&q=75 2x"
    style="color: transparent; height: 18rem; width: 100%;"
    width="400"
  />
</div>
`;

exports[`Profile badge size lg matches unstyled snapshot 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <img
    alt="Meeple Party"
    class=""
    data-nimg="1"
    decoding="async"
    height="300"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=828&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg.jpg&w=828&q=75 2x"
    style="color: transparent;"
    width="400"
  />
</div>
`;

exports[`Profile badge size md matches snapshot 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <img
    alt="Meeple Party"
    class=""
    data-nimg="1"
    decoding="async"
    height="72"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=256&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=96&q=75 1x, /_next/image?url=%2Fimg.jpg&w=256&q=75 2x"
    style="color: transparent; height: 6rem; width: 100%;"
    width="96"
  />
</div>
`;

exports[`Profile badge size md matches snapshot with additional class names 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <img
    alt="Meeple Party"
    class="test class name"
    data-nimg="1"
    decoding="async"
    height="72"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=256&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=96&q=75 1x, /_next/image?url=%2Fimg.jpg&w=256&q=75 2x"
    style="color: transparent; height: 6rem; width: 100%;"
    width="96"
  />
</div>
`;

exports[`Profile badge size md matches unstyled snapshot 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <img
    alt="Meeple Party"
    class=""
    data-nimg="1"
    decoding="async"
    height="72"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=256&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=96&q=75 1x, /_next/image?url=%2Fimg.jpg&w=256&q=75 2x"
    style="color: transparent;"
    width="96"
  />
</div>
`;

exports[`Profile badge size sm matches snapshot 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <img
    alt="Meeple Party"
    class=""
    data-nimg="1"
    decoding="async"
    height="18"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
    style="color: transparent; height: 3rem; width: 100%;"
    width="24"
  />
</div>
`;

exports[`Profile badge size sm matches snapshot with additional class names 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <img
    alt="Meeple Party"
    class="test class name"
    data-nimg="1"
    decoding="async"
    height="18"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
    style="color: transparent; height: 3rem; width: 100%;"
    width="24"
  />
</div>
`;

exports[`Profile badge size sm matches unstyled snapshot 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <img
    alt="Meeple Party"
    class=""
    data-nimg="1"
    decoding="async"
    height="18"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
    style="color: transparent;"
    width="24"
  />
</div>
`;

exports[`Profile badge size xs matches snapshot 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <img
    alt="Meeple Party"
    class=""
    data-nimg="1"
    decoding="async"
    height="12"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
    style="color: transparent; height: 2ex; width: 100%;"
    width="16"
  />
</div>
`;

exports[`Profile badge size xs matches snapshot with additional class names 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <img
    alt="Meeple Party"
    class="test class name"
    data-nimg="1"
    decoding="async"
    height="12"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
    style="color: transparent; height: 2ex; width: 100%;"
    width="16"
  />
</div>
`;

exports[`Profile badge size xs matches unstyled snapshot 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <img
    alt="Meeple Party"
    class=""
    data-nimg="1"
    decoding="async"
    height="12"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
    style="color: transparent;"
    width="16"
  />
</div>
`;
